src/Psalm/Type/Atomic/TString.php

```diff
@@ -3,6 +3,17 @@
 
 class TString extends Scalar
```

Comment:
I've been thinking about something similar. What's the reason for choosing this implementation over `TLiteralString extends Scalar` (something that can cast to/from a TString).

E.g. if a variable could have three possible string values, it could be a union type of 3 TLiteralString types instead of a union type with a single type (with 3 possible strings), which would seem like it would make merging branches easier

Reply:
No good reason - I think `TLiteralString` is a great classname.